cwms_radar_api/src/main/java/cwms/radar/data/dto/TimeSeries.java

```diff
@@ -52,38 +53,63 @@ public class TimeSeries extends CwmsDTOPaginated {
     @XmlElement(name = "office-id")
     String officeId;
 
-    @Schema(description = "The units of the time series data")
+    @Schema(description = "The units of the time series data",required = true)
     String units;
 
     @XmlJavaTypeAdapter(DurationAdapter.class)
     @JsonFormat(shape = Shape.STRING)
-    @Schema(description = "The interval of the time-series, in ISO-8601 duration format")
+    @Schema(
+        accessMode = AccessMode.READ_ONLY,
+        format = "Java Duration",
+        description = "The interval of the time-series, in ISO-8601 duration format"
+        )
     Duration interval;
 
     @XmlJavaTypeAdapter(ZonedDateTimeAdapter.class)
     @JsonFormat(shape = Shape.STRING, pattern = ZONED_DATE_TIME_FORMAT)
-    @Schema(description = "The requested start time of the data, in ISO-8601 format with offset and timezone ('" + ZONED_DATE_TIME_FORMAT + "')")
+    @Schema(
+        accessMode = AccessMode.READ_ONLY,
+        description = "The requested start time of the data, in ISO-8601 format with offset and timezone ('" + ZONED_DATE_TIME_FORMAT + "')"
+    )
     ZonedDateTime begin;
 
     @XmlJavaTypeAdapter(ZonedDateTimeAdapter.class)
     @JsonFormat(shape = Shape.STRING, pattern = ZONED_DATE_TIME_FORMAT)
-    @Schema(description = "The requested end time of the data, in ISO-8601 format with offset and timezone ('" + ZONED_DATE_TIME_FORMAT + "')")
+    @Schema(
+        accessMode = AccessMode.READ_ONLY,
+        description = "The requested end time of the data, in ISO-8601 format with offset and timezone ('" + ZONED_DATE_TIME_FORMAT + "')"
+    )
     ZonedDateTime end;
 
     @XmlElementWrapper
     @XmlElement(name="record")
     // Use the array shape to optimize data transfer to client
     @JsonFormat(shape=JsonFormat.Shape.ARRAY)
-    @Schema(description = "List of retrieved time-series values")
+    @ArraySchema(
+        schema = @Schema(
+            description = "List of retrieved time-series values",
+            implementation = Record.class
+        )
+    )
     List<Record> values;
```

Comment:
I don't appreciate the usefulness of the @ArraySchema annotation.  I'm not saying this sarcastically, I really don't know what it does.  I tried these changes and one of the things that changed was that the $ref field on TimeSeries.items went away.  That $ref field makes the codegen Adam is using map our List to a Record[].  Maybe there are more annotations we can add to make the codegen work once again with the ArraySchema.  I don't know.  It seems like telling swagger that its an array of Record objects could only help but the schema that gets generated doesn't contain anything linking the field to Record.  The generated schema does describe the shape of the array items so it seems like sufficiently smart codegen could still make it work.  Once these changes get merged we should have Adam redo codegen.

![Screenshot 2022-09-02 124910](https://user-images.githubusercontent.com/89810919/188227474-23974547-9388-4593-b95b-c08f09d46401.png)

Reply:
I'll have to dig into that as well. It was necessary to get the example to show up correctly in the swagger ui, otherwise it doesn't render for the user correctly.
However one can modify the returned schema. so I could probably work something so that it's only rendered that was for the swagger-ui.

But hopefully there's a way to fix the ref, maybe now that I have an example that gets render it's not required.